### PR TITLE
[compiler-rt][nsan] Modify NSAN_CFLAGS setting

### LINKED ISSUE #98767: [compiler-rt][nsan] Undefined reference to vtable for ... ...
Consider the simple example
`#include <stddef.h>
int main(int argc, char *av[])
{
return 0;
}`

and build it with trunk clang:
`clang test.c -fsanitize=numerical`

then the following errors occur
`/usr/bin/ld: /usr/lib/clang/19/lib/x86_64-unknown-linux-gnu/libclang_rt.nsan.a ... ...
undefined reference to vtable for __cxxabiv1::__class_type_info
undefined reference to vtable for __cxxabiv1::__si_class_type_info
...
...
`

This error occurs because libclang_rt.nsan.a objects are built without `-fno-rtti` flag.
OS: ArchLinux

### PR DESCRIPTION
**Pull request by chestnykh**
Other runtime libraries build scripts
like asan, dfsan, msan etc.
use `-fno-rtti` flag to avoid vtables in the runtime library
so add `-fno-rtti` to `NSAN_CFLAGS`

Also remove unneeded adding `-fPIC` to `NSAN_CFLAGS` that was done in the root CMakeLists.txt of compiler-rt

Fix #98767

### CODE REVIEW

**Review by alexander-shaposhnikov — APPROVED** [review]

**Comment by chestnykh** [comment]
@vitalybuka FYI I don't have commit access so please merge this PR when it is ready

**Pull request merged**
